lib/train/transports/gcp.rb

```diff
@@ -0,0 +1,101 @@
+# encoding: utf-8
+
+require 'train/plugins'
+require 'google/apis'
+require 'google/apis/cloudresourcemanager_v1'
+require 'google/apis/compute_v1'
+require 'google/apis/storage_v1'
+require 'google/apis/iam_v1'
+require 'googleauth'
+require 'JSON'
+
+module Train::Transports
+  class Gcp < Train.plugin(1)
+    name 'gcp'
+
+    # GCP will look automatically for the below env var for service accounts etc. :
+    option :google_application_credentials, required: false, default: ENV['GOOGLE_APPLICATION_CREDENTIALS']
+    # see https://cloud.google.com/docs/authentication/production#providing_credentials_to_your_application
+    # In the absence of this, the client is expected to have already set up local credentials via:
+    #   $ gcloud auth application-default login
+    #   $ gcloud config set project <project-name>
+    # GCP projects can have default regions / zones set, see:
+    # https://cloud.google.com/compute/docs/regions-zones/changing-default-zone-region
+    # can also specify project via env var:
+    option :google_cloud_project, required: false, default: ENV['GOOGLE_CLOUD_PROJECT']
+
+    def connection(_ = nil)
+      @connection ||= Connection.new(@options)
+    end
+
+    class Connection < BaseConnection
+      def initialize(options)
+        super(options)
+
+        # additional GCP platform metadata
+        release = Gem.loaded_specs['google_cloud']
+        @platform_details = { release: "google-cloud-v#{release}" }
+
+        # Initialize the client object cache
+        @cache_enabled[:api_call] = true
+        @cache[:api_call] = {}
+
+        connect
+      end
+
+      def platform
+        direct_platform('gcp', @platform_details)
+      end
+
+      # Instantiate some named classes for ease of use
+      def gcp_compute_client
```

Comment:
if your going to make helper methods can you use the `gcp_client` for them all? That way the logic is in one spot.
```
def gcp_iam_client
   gcp_client(Google::Apis::IamV1::IamService)
end
```

Reply:
Good shout! I'll update that now.